src/Missings.jl

```diff
@@ -208,6 +208,104 @@ missing
 """
 passmissing(f) = PassMissing{Core.Typeof(f)}(f)
 
+struct SpreadMissings{F} <: Function
+    f::F
+end
+
+function (f::SpreadMissings{F})(x) where {F}
+    @assert x isa AbstractVector
+
+    if x isa AbstractVector{>:Missing}
+        nonmissinginds = collect(eachindex(skipmissing(x)))
+        res = f.f(view(x, nonmissinginds))
+
+        out = Vector{Union{Missing, eltype(res)}}(missing, length(x))
+        out[nonmissinginds] .= res
+
+        return out
+    else
+        return f.f(x)
+    end
+end
+
+function (f::SpreadMissings{F})(xs...) where {F}
+    if any(x -> x isa AbstractVector{>:Missing}, xs)
+        vecs = Base.filter(x -> x isa AbstractVector, xs)
+        s = skipmissings(vecs...)
```

Comment:
As we discussed somewhere in a DataFrames issue, I don't think using `skipmissings` here is really efficient: it's going to require two passes over the data: one during the call to `f`, and one for `collect(eachindex(first(s)))`. Since you do the latter step, better use the result to create `SubArray` views of arguments, which are faster than `skipmissings` and more convenient for users.